pkg/command/service/generate.go

```diff
@@ -75,51 +85,35 @@ kperf service generate —n 500 —interval 20 —batch 20 --min-scale 0 --max-s
 				}
 				if start > 0 && end > 0 && start <= end {
 					for i := start; i <= end; i++ {
-						nsRangeMap[fmt.Sprintf("%s-%d", nsPrefix, i)] = true
+						nsNameList = append(nsNameList, fmt.Sprintf("%s-%d", nsPrefix, i))
 					}
 				} else {
 					return errors.New("failed to parse namespace range")
 				}
+			} else if ns != "" {
+				nsNameList = append(nsNameList, ns)
 			}
 
-			if err != nil {
-				return err
+			if len(nsNameList) == 0 {
+				return fmt.Errorf("no namespace found %s", nsPrefix)
 			}
-			nsNameList := []string{}
-			if ns != "" {
-				nss, err := p.ClientSet.CoreV1().Namespaces().Get(context.TODO(), ns, metav1.GetOptions{})
-				if err != nil {
-					return err
-				}
-				nsNameList = append(nsNameList, nss.Name)
-			} else if nsPrefix != "" {
-				nsList, err := p.ClientSet.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
-				if err != nil {
-					return err
-				}
-				if len(nsList.Items) == 0 {
-					return fmt.Errorf("no namespace found with prefix %s", nsPrefix)
-				}
-				if len(nsRangeMap) >= 0 {
-					for i := 0; i < len(nsList.Items); i++ {
-						if _, exists := nsRangeMap[nsList.Items[i].Name]; exists {
-							nsNameList = append(nsNameList, nsList.Items[i].Name)
-						}
-					}
-				} else {
-					for i := 0; i < len(nsList.Items); i++ {
-						if strings.HasPrefix(nsList.Items[i].Name, nsPrefix) {
-							nsNameList = append(nsNameList, nsList.Items[i].Name)
-						}
-					}
-				}
 
-				if len(nsNameList) == 0 {
-					return fmt.Errorf("no namespace found with prefix %s", nsPrefix)
+			// Check if namespace exists, in NOT, return error
+			for _, ns := range nsNameList {
+				_, err := p.ClientSet.CoreV1().Namespaces().Get(context.TODO(), ns, metav1.GetOptions{})
+				if err != nil && apierrors.IsNotFound(err) {
+					return errors.Errorf("namespace %s not found, please create one", ns)
+				} else if err != nil {
+					return errors.Wrap(err, "failed to get namespace")
 				}
-			} else {
-				return fmt.Errorf("both ns and nsPrefix are empty")
 			}
+
+			restConfig, err := p.RestConfig()
+			ksvcClient, err = servingv1client.NewForConfig(restConfig)
```

Comment:
After I refine serving client initialization, you can use `ksvcClient, err := p.NewServingClient()` to avoid initial it again.

Reply:
@gyohuangxin Sure, will cover by a rebase.